go/cmd/dolt/commands/merge.go

```diff
@@ -93,146 +99,274 @@ func (cmd MergeCmd) Exec(ctx context.Context, commandStr string, args []string,
 		defer closeFunc()
 	}
 
-	if apr.ContainsAll(cli.SquashParam, cli.NoFFParam) {
-		cli.PrintErrf("error: Flags '--%s' and '--%s' cannot be used together.\n", cli.SquashParam, cli.NoFFParam)
+	ok := validateDoltMergeArgs(apr, usage, cliCtx)
```

Comment:
Before we validate the args, verify that the queryist is a SQLEnginer instance. If it's not, then this is a remote usage, and we should print an error telling them to stop their local server.